src/test/java/uk/gov/ons/census/action/model/repository/CaseRepositoryIT.java

```diff
@@ -0,0 +1,142 @@
+package uk.gov.ons.census.action.model.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.data.jpa.domain.Specification.where;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.persistence.criteria.CriteriaBuilder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.census.action.model.entity.Case;
+
+@ActiveProfiles("test")
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class CaseRepositoryIT {
+
+  private static final String TEST_ACTION_PLAN_ID;
+  private static final Map<String, List<String>> TEST_CLASSIFIERS;
+
+  @Autowired private CaseRepository caseRepository;
+
+  static {
+    TEST_ACTION_PLAN_ID = "ddbd26c2-15af-4055-aab9-591b3735b8d3";
+
+    TEST_CLASSIFIERS = new HashMap<>();
+    TEST_CLASSIFIERS.put(
+        "treatmentCode",
+        Arrays.asList(
+            "HH_LF2R3AE",
+            "HH_LF3R3BE",
+            "HH_LF3R3AE",
+            "HH_LF2R1E",
+            "HH_LFNR2E",
+            "HH_LFNR3AE",
+            "HH_LF2R3BE",
+            "HH_LF3R2E",
+            "HH_LF2R2E",
+            "HH_LF3R1E"));
+  }
+
+  @Transactional
+  @Test
+  @Sql(scripts = {"/data-test-10of10-unreceipted.sql"})
+  public void shouldRetrieveTenCasesWhenNoneReceiptedAndWithoutClassifiers() {
+    int expectedCaseCount = 10;
+
+    Specification<Case> expectedSpecification = getSpecificationWithoutClassifiers();
+
+    List<Case> cases = caseRepository.findAll(expectedSpecification);
+
+    assertThat(cases.size()).isEqualTo(expectedCaseCount);
+  }
+
+  @Transactional
+  @Test
+  @Sql(scripts = {"/data-test-03of10-receipted.sql"})
```

Comment:
I'm not a huge fan of native SQL. I really think we should avoid it

Reply:
Removed and using EasyRandom instead.